### PR TITLE
Chore/improve devx

### PR DESCRIPTION
**Pull request by jkoenig134**
Some huge devX improvements:
- there is now a [root level `tsconfig.json`](https://github.com/nmshd/runtime/blob/chore/improve-devx/tsconfig.json) that lets us compile every project at once. This helps for example adding:
- [root level `build` and `build:watch`](https://github.com/nmshd/runtime/blob/chore/improve-devx/package.json#L17-L18) - are our new scripts to build all the libraries at once. changed files are recompiled incrementally with the watching script
- [more docs](https://github.com/nmshd/runtime/blob/chore/improve-devx/README_dev.md) 😃 
- easier development for iql (or more precise easier development when using iql but not touching the peggy definition)

### CODE REVIEW

**Review by sebbi08 — APPROVED** [review]

**Comment by jkoenig134** [comment]
as this affects the devx of many people I'd wait for more approvals, especially by @Milena-Czierlinski and @slavistan which already worked with the monorepo - please share your thoughts!

File: README_dev.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 1. Download and install [Node JS](https://nodejs.org/en/download/)
 2. run `npm i`
+3. run `npm build:watch`
```
Comment (Milena-Czierlinski) [inline README_dev.md:5]: missing `run`

Also, do I actually need to call this. For me another shell opened automatically.

File: README_dev.md [hunk 2]
```diff
@@ -17,6 +18,14 @@ This workspace provide two types of scripts:
 
     additionally you can run a script in all workspaces by `npm run -ws --if-present <scriptname>`
 
+## Linting
+
+-   Typescript
+    -   each package provides its own typescript linting script. You can run it by `npm run lint:ts` from the package directory or by `npm run -w packages/<package> lint` from the root of the project
```
Comment (Milena-Czierlinski) [inline README_dev.md:24]: npm run lint:ts**c** (also below)

Also for the second occurrence here.

File: README_dev.md [hunk 3]
```diff
@@ -17,6 +18,14 @@ This workspace provide two types of scripts:
...
+    -   the root project also provides a script that lints all packages. You can run it by `npm run lint:ts` from the root of the project
+-   Prettier: simply run `npm run prettier` from the root of the project
```
Comment (Milena-Czierlinski) [inline README_dev.md:26]: npm run **lint:**prettier

File: README_dev.md [hunk 4]
```diff
@@ -17,6 +18,14 @@ This workspace provide two types of scripts:
...
+-   ESLint: simply run `npm run lint:eslint` from the root of the project
```
Comment (Milena-Czierlinski) [inline README_dev.md:27]: This fails on my computer

FATAL ERROR: Ineffective mark-compacts near heap limit Allocation failed - JavaScript heap out of memory

File: .vscode/tasks.json
Comment (Milena-Czierlinski) [inline .vscode/tasks.json:1]: When changing a function, the compile task automatically shows me the error it causes in the same workspace, which I really like. However, errors in other workspaces that are caused by it are not shown. As soon as I open the file in the other workspace, the respective piece of code is emphasized. But I'd love for the compile task to show me earlier, like for the places the function is used in the same workspace.
Is it possible to implement this as well?

Comment (jkoenig134) [reply .vscode/tasks.json:1]: We have to talk about your usecase of having multiple workspaces.

File: README_dev.md (hunk 4 above)
Comment (jkoenig134) [reply README_dev.md:27]: oh no

File: README_dev.md [hunk 5]
```diff
@@ -21,9 +21,9 @@ This workspace provide two types of scripts:
 ## Linting
 
 -   Typescript
-    -   each package provides its own typescript linting script. You can run it by `npm run lint:ts` from the package directory or by `npm run -w packages/<package> lint` from the root of the project
-    -   the root project also provides a script that lints all packages. You can run it by `npm run lint:ts` from the root of the project
--   Prettier: simply run `npm run prettier` from the root of the project
+    -   each package provides its own typescript linting script. You can run it by `npm run lint:tsc` from the package directory or by `npm run -w packages/<package> lint` from the root of the project
```
Comment (Milena-Czierlinski) [inline README_dev.md:24]: The second occurrence here misses the `:tsc` -> `npm run -w packages/<package> lint:tsc`

File: README_dev.md (hunk 4 above)
Comment (jkoenig134) [reply README_dev.md:27]: fixed - wsl had not enough memory

**Review by Milena-Czierlinski — APPROVED** [review]

**Pull request merged**